### PR TITLE
o/assertstate: fix handling of validation set tracking update in enforcing mode

### PR DESCRIPTION
**Pull request by stolowski**
Fix handling of validation set tracking update in enforcing mode when pinning/un-pinning or pinning to another sequence number, including a lower one. Make sure Current points to the latest sequence available in such cases.
   
Fixes https://bugs.launchpad.net/snapd/+bug/1969216

### CODE REVIEW

File: overlord/assertstate/validation_set_tracking.go [hunk 1]
```diff
@@ -156,11 +158,21 @@ func EnforcedValidationSets(st *state.State) (*snapasserts.ValidationSets, error
 	db := DB(st)
 	sets := snapasserts.NewValidationSets()
 
+	if extraVs != nil {
+		sets.Add(extraVs)
```
Comment (pedronis) [inline overlord/assertstate/validation_set_tracking.go:162]: and this

File: overlord/assertstate/assertstate.go [hunk 2]
```diff
@@ -659,8 +654,9 @@ func validationSetAssertionForEnforce(st *state.State, accountID, name string, s
 	vs, err = getSpecificSequenceOrLatest(db, headers)
 
 	checkForConflicts := func() error {
-		if err := valsets.Add(vs); err != nil {
-			return fmt.Errorf("internal error: cannot check validation sets conflicts: %v", err)
+		valsets, err := EnforcedValidationSets(st, vs)
```
Comment (pedronis) [inline overlord/assertstate/assertstate.go:657]: this reads nicely

**Review by pedronis — APPROVED** [review]
thank you, lgtm, comment about unit tests

File: overlord/assertstate/assertstate.go [hunk 3]
```diff
@@ -600,23 +600,18 @@ func validationSetAssertionForMonitor(st *state.State, accountID, name string, s
 // checks if it's not in conflict with existing validation sets in enforcing mode
 // (all currently tracked validation set assertions get refreshed), and if they
 // are valid for installed snaps.
```
Comment (mardy) [inline overlord/assertstate/assertstate.go:602]: Please document what the `current` return value is.

**Review by mardy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**